service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -30,6 +40,74 @@ public Claim insertClaim(Claim claim) {
     return claim;
   }
 
+  @Override
+  public AssessmentResult insertAssessmentResult(
+      UUID id,
+      String assessmentResult,
+      String veteranIcn,
+      String claimSubmissionId,
+      String idType,
+      String diagnosticCode)
+      throws NoSuchFieldException {
+    ClaimEntity claimEntity =
+        claimRepository
+            .findById(id)
+            .orElseThrow(
+                () ->
+                    new NoSuchFieldException(
+                        "Could not match claim ID {" + id.toString() + "} in DB."));
+    Map<String, Object> evidence = new HashMap<>();
+    try {
+      evidence = objMapper.readValue(assessmentResult, Map.class);
+      if (evidence.isEmpty()) {
+        throw new NoSuchFieldException("Could not map assesmentResult, evidence summary is null.");
+      }
+    } catch (Exception e) {
+      log.error("Could not map assesmentResult, evidence summary is null.");
+    }
+    Map<String, Object> defaultEvidence = new HashMap<>();
+    defaultEvidence.put("medicationsCount", 0);
+    Map summary = (Map) evidence.getOrDefault("evidenceSummary", defaultEvidence);
+    int count = 0;
+    if (summary != null) {
+      for (Object value : summary.values()) {
+        count += (Integer) value;
+      }
+    }
+    AssessmentResultEntity assessmentResultEntity = new AssessmentResultEntity();
+    assessmentResultEntity.setEvidenceCount(count);
+    assessmentResultEntity.setEvidenceCountSummary(summary);
+    ContentionEntity contention = findContention(claimEntity, diagnosticCode);
+    for (AssessmentResultEntity ar : contention.getAssessmentResults()) {
+      if (ar.getContention().getId() == contention.getId()) {
+        log.info("Assessment result already populated, continuing.");
+        break;
+      }
```

Comment:
I do not understand this for loop. break just gets out of the for loop and continues. Is this just for the message? Otherwise it does not change the flow at all.